js/scorm/wrapper.js

```diff
@@ -605,77 +605,60 @@ class ScormWrapper {
   }
 
   recordInteractionScorm12(id, response, correct, latency, type) {
-
+    const children = this.getValue('cmi.interactions._children');
+    if (this.isUnsupportedLastError() || !children.includes('id') || !this.isSupported('cmi.interactions._count')) {
+      this.logUnsupported('cmi.interactions');
+      return;
+    }
     id = id.trim();
-
     const cmiPrefix = `cmi.interactions.${this.getInteractionCount()}`;
-
     this.setValue(`${cmiPrefix}.id`, id);
-    this.setValue(`${cmiPrefix}.type`, type);
-    this.setValue(`${cmiPrefix}.student_response`, response);
-    this.setValue(`${cmiPrefix}.result`, correct ? 'correct' : 'wrong');
-    if (latency !== null && latency !== undefined) this.setValue(`${cmiPrefix}.latency`, this.convertToSCORM12Time(latency));
-    this.setValue(`${cmiPrefix}.time`, this.getCMITime());
+    children.includes('type') ? this.setValue(`${cmiPrefix}.type`, type) : this.logUnsupported(`${cmiPrefix}.type`);
+    children.includes('student_response') ? this.setValue(`${cmiPrefix}.student_response`, response) : this.logUnsupported(`${cmiPrefix}.student_response`);
+    children.includes('result') ? this.setValue(`${cmiPrefix}.result`, correct ? 'correct' : 'wrong') : this.logUnsupported(`${cmiPrefix}.result`);
+    children.includes('latency') && latency != null ? this.setValue(`${cmiPrefix}.latency`, this.convertToSCORM12Time(latency)) : this.logUnsupported(`${cmiPrefix}.latency`);
+    children.includes('time') ? this.setValue(`${cmiPrefix}.time`, this.getCMITime()) : this.logUnsupported(`${cmiPrefix}.time`);
```

Comment:
Would the following be preferable for readability so the ternary confidtion isn't all on one line?
```suggestion
    children.includes('type') ? this.setValue(`${cmiPrefix}.type`, type)
      : this.logUnsupported(`${cmiPrefix}.type`);
    children.includes('student_response') ? this.setValue(`${cmiPrefix}.student_response`, response)
      : this.logUnsupported(`${cmiPrefix}.student_response`);
    children.includes('result') ? this.setValue(`${cmiPrefix}.result`, correct ? 'correct' : 'wrong')
      : this.logUnsupported(`${cmiPrefix}.result`);
    children.includes('latency') && latency != null ? this.setValue(`${cmiPrefix}.latency`, this.convertToSCORM12Time(latency))
      : this.logUnsupported(`${cmiPrefix}.latency`);
    children.includes('time') ? this.setValue(`${cmiPrefix}.time`, this.getCMITime())
      : this.logUnsupported(`${cmiPrefix}.time`);
```